action/protocol/execution/evm/evmstatedbadapter.go

```diff
@@ -892,10 +919,16 @@ func (stateDB *StateDBAdapter) clear() {
 	stateDB.suicideSnapshot = nil
 	stateDB.preimages = nil
 	stateDB.preimageSnapshot = nil
+	stateDB.logsSnapshot = nil
```

Comment:
Once the memory is unreachable, it will be freed.
No need to set it to `nil`

Reply:
yes, reason for this:
1. this is for code clarity (clearly tells reader the variable is not needed anymore), follow the existing code
2. in a sense help the runtime/GC to know immediately this memory is no longer needed and can be GC'ed, similar to https://github.com/iotexproject/iotex-core/commit/90badd6a83f9ff1b4e8b9addf58a83503f67e36e#diff-aa7bcf8c732d6ec1b1b2d3a3324071563c4c913f007f3298bf8f7e1f7c78aaa5R49

Reply:
Unlike C++, Go is automatic memory management. The GC won't invoked immediately when the reference is removed.

Reply:
right ... again the reasons explained above